### PR TITLE
add ide settings permission and enable it for devops role

### LINKED ISSUE #1948: add ide feature flag
Only if it is enabled a user can pick an alternative IDE, otherwise Theia shoulde be used always.

### PR DESCRIPTION
**Pull request by akosyakov**
- [x] /werft ws-feature-flags=registry_facade
- [x] /werft https

#### What it does

- fix #1948: `ide-settings` permission is added which is enabled by default for `devops` role. This permission allow to configure the default ide image as well as control workspace specific ide image from `.gitpod.yml`. Otherwise such settings are ignored.

#### How to test

- check then Theia is used even if .gitpod.yml contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com//#https://github.com/akosyakov/go-gin-app
- go to your settings page and check that `IDE Settings` section is not visible: https://ak-ide-permission.staging.gitpod-dev.com/settings/
- grant `devops` role to your user, i.e. run a query like: `update d_b_user set rolesOrPermissions = '["admin","devops"]' where name = 'akosyakov';`
- check that Code is used if .gitpod.yml contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com//#https://github.com/akosyakov/go-gin-app
- check that Theia is used if .gitpod.yml does NOT contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com/#https://github.com/gitpod-io/go-gin-app
- go to your settings page and switch the default ide to `Code` instead of `Theia`: https://ak-ide-permission.staging.gitpod-dev.com/settings/
- check that Code is used if .gitpod.yml does NOT contains `ide: code`: http://ak-ide-permission.staging.gitpod-dev.com/#https://github.com/gitpod-io/go-gin-app

### CODE REVIEW

File: components/dashboard/public/schemas/gitpod-schema.json [hunk 1]
```diff
@@ -5,16 +5,10 @@
     "type": "object",
     "properties": {
         "ide": {
-          "oneOf":[
```
Comment (akosyakov) [inline components/dashboard/public/schemas/gitpod-schema.json:8]: It does not work and breaks schema validation. I am not sure how allow any string or enum, for now it is just theia or code to provide content assist.

Comment (akosyakov) [reply components/dashboard/public/schemas/gitpod-schema.json:8]: it should be `anyOf`: https://github.com/json-schema-org/json-schema-spec/issues/665#issuecomment-432072391

**Review by csweichel — APPROVED** [review]
works like a charm and code LGTM

**Pull request merged**